pkg/device42/customer.go

```diff
@@ -0,0 +1,72 @@
+package device42
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+
+	"github.com/asecurityteam/ipam-facade/pkg/domain"
+)
+
+type customersResponse struct {
+	Customers []customer `json:"Customers"`
+}
+
+type customer struct {
+	Contacts     []contact    `json:"Contacts"`
+	CustomFields customFields `json:"Custom Fields"`
+	ContactInfo  string       `json:"contact_info"`
+	DevicesURL   string       `json:"devices_url"`
+	Groups       string       `json:"groups"`
+	ID           int          `json:"id"`
+	Name         string       `json:"name"`
+	Notes        string       `json:"notes"`
+	SubnetsURL   string       `json:"subnets_url"`
+}
+
+type contact struct {
+	Address string `json:"address"`
+	Email   string `json:"email"`
+	Name    string `json:"name"`
+	Phone   string `json:"phone"`
+	Type    string `json:"type"`
+}
+
+// Device42CustomerFetcher fetches customer data from Device42
+type Device42CustomerFetcher struct {
+	Client *http.Client
+	Host   *url.URL
+}
+
+// FetchCustomers fetches customers from IPAM
+func (d *Device42CustomerFetcher) FetchCustomers(ctx context.Context) ([]domain.Customer, error) {
```

Comment:
Task: Somewhere in here you should check the response status code. As it stands, even if Device42 returns an error, as long as the body will unmarshal, no error will be returned from this function

Reply:
For reference: https://github.com/asecurityteam/nexpose-scan-notifier/pull/6